src/components/afm/DataSourceProvider.tsx

```diff
@@ -127,13 +134,38 @@ export function dataSourceProvider<T>(
             this.subject.unsubscribe();
         }
 
+        // updateTotals must be called everytime totals are updated
+        // to check if we need to reload datasource to get new native totals
+        // @param nextTotals lists all requested totals
+        public updateTotals(nextTotals: AFM.ITotalItem[]) {
+            const nativeTotals = getNativeTotals(nextTotals);
+            const afm: AFM.IAfm = this.state.dataSource.getAfm();
+            if (
+                // native totals requested
+                nativeTotals.length > 0
+                && (
+                    // nativeTotals not defined in afm.nativeTotals
+                    !afm.nativeTotals
+                    // or required native totals not listed among defined afm.nativeTotals
+                    || nativeTotals.some(total => !afm.nativeTotals || !afm.nativeTotals.find(
+                        afmNativeTotal => isEqual(afmNativeTotal, total)
+                    ))
+                )
```

Comment:
Maybe a nit pick, but consider instead of

```ts
if (
    // native totals requested
    nativeTotals.length > 0
    && (
        // nativeTotals not defined in afm.nativeTotals
        !afm.nativeTotals
        // or required native totals not listed among defined afm.nativeTotals
        || nativeTotals.some(total => !afm.nativeTotals || !afm.nativeTotals.find(
            afmNativeTotal => isEqual(afmNativeTotal, total)
        ))
    )
) {
    //...
}
```

using something like

```ts
const nativeTotalsRequested = nativeTotals.length > 0;
const hasNativeTotals = !!afm.nativeTotals;
const hasUnlistedNativeTotals = nativeTotals.some(total => !afm.nativeTotals || !afm.nativeTotals.find(
    afmNativeTotal => isEqual(afmNativeTotal, total)
));
if (nativeTotalsRequested && (!hasNativeTotals || hasUnlistedNativeTotals)) {
    //...
}
```

to avoid unnecessary comments